server/src/test/java/org/elasticsearch/index/query/RandomQueryBuilder.java

```diff
@@ -48,7 +47,9 @@ public static QueryBuilder createQuery(Random r) {
             case 1:
                 return new TermQueryBuilderTests().createTestQueryBuilder();
             case 2:
-                return new ExistsQueryBuilder(randomAlphaOfLength(5));
+                // We make sure this query has no types to avoid deprecation warnings in the
+                // tests that use this method.
+                return new IdsQueryBuilderTests().createTestQueryBuilder().types(new String[0]);
```

Comment:
nit: using String.EMPTY_ARRAY would save another array creation. But only change this if there are other changes needed to this PR, otherwise its perfectly fine.

Reply:
👍 good to know about this.